### ISSUE TITLE
Add Gradient Norm to Optimizer Metrics

### ISSUE THREAD

**Issue by maitchison** [thinking-machines-lab/tinker-feedback#96]
# Request

Add pre-clipped global gradient L2 norm during optimizer step to `OptimStepResponse.metrics` 

# Justification

Gradient norm can be a useful thing to monitor, especially in relation to the gradient clipping threshold. We currently have no visibility into the gradient norms, as these are done server side.
